src/sentry/integrations/api/serializers/rest_framework/data_forwarder.py

```diff
@@ -0,0 +1,206 @@
+        # optional_fields = ["message_group_id", "s3_bucket"]
+
+        missing_fields = [field for field in required_fields if field not in config]
+        if missing_fields:
+            raise ValidationError(f"Missing required SQS fields: {', '.join(missing_fields)}")
+
+        errors = []
+
+        # SQS URL format: https://sqs.region.amazonaws.com/account/queue-name
+        queue_url = config.get("queue_url")
+        sqs_url_pattern = (
+            r"^https://sqs\.[a-z0-9\-]+\.amazonaws\.com/\d+/[a-zA-Z0-9\-_/]+(?:\.fifo)?$"
+        )
+        if not isinstance(queue_url, str) or not re.match(sqs_url_pattern, queue_url):
+            errors.append(
+                "queue_url must be a valid SQS URL format: "
+                "https://sqs.region.amazonaws.com/account/queue-name"
+            )
+
+        aws_region_pattern = r"^[a-z0-9\-]+$"
+        region = config.get("region")
+        if not isinstance(region, str) or not re.match(aws_region_pattern, region):
+            errors.append("region must be a valid AWS region format")
+
+        access_key = config.get("access_key")
+        secret_key = config.get("secret_key")
+
+        if not isinstance(access_key, str) or access_key.strip() == "":
+            errors.append("access_key must be a non-empty string")
+
+        if not isinstance(secret_key, str) or secret_key.strip() == "":
+            errors.append("secret_key must be a non-empty string")
+
+        message_group_id = config.get("message_group_id")
+        if message_group_id is not None and not isinstance(message_group_id, str):
+            errors.append("message_group_id must be a string or null")
+
+        if isinstance(queue_url, str) and queue_url.endswith(".fifo") and not message_group_id:
+            errors.append("message_group_id is required for FIFO queues")
+
+        s3_bucket = config.get("s3_bucket")
+        if s3_bucket is not None:
+            if not isinstance(s3_bucket, str) or s3_bucket.strip() == "":
+                errors.append("s3_bucket must be a non-empty string")
+            else:
+                s3_bucket_pattern = r"^[a-z0-9\-\.]+$"
+                if not re.match(s3_bucket_pattern, s3_bucket):
+                    errors.append("s3_bucket must be a valid S3 bucket name")
+
+        if errors:
+            raise ValidationError(errors)
+
+        return config
+
+    def _validate_segment_config(self, config: dict) -> dict:
+        if "write_key" not in config:
+            raise ValidationError("Missing required Segment fields: write_key")
+
+        segment_write_key_pattern = r"^[a-zA-Z0-9_\-]+$"
+        write_key = config.get("write_key")
+        if not isinstance(write_key, str) or not re.match(segment_write_key_pattern, write_key):
+            raise ValidationError("write_key must be a valid Segment write key format")
+
+        return config
+
+    def _validate_splunk_config(self, config: dict) -> dict:
```

Comment:
If the output we're getting out of the validator is a known form I think we should type it so/specify in the function + hint. Here's an [example](https://github.com/getsentry/sentry/blob/leander/eco-1102-email-preview-in-debugger-app/src/sentry/api/serializers/rest_framework/groupsearchview.py#L12) of existing `TypedDict`s I think would help with readability and potentially validation here. But I think maybe something like the below maybe?

```
class SplunkConfig(TypedDict):
     instance_URL: str
     index: str
     source: str
     token: str
```